sphinx/source/developers/consensus.rst

```diff
@@ -0,0 +1,41 @@
+Consensus Protocols
+===================
+
+In order to guarantee system reliability changes to the key-value store must be agreed by a number of nodes before being applied. The number of nodes that need to agree on a change before it is applied depends on the consensus protocol being run.
+
+The default consensus protocol for CCF is a version of Raft which provides Crash Fault Tolerance.
+
+We are currently working on providing PBFT as an additional consensus option. PBFT provides Byzantine Fault Tolerance.
+
+.. warning:: Currently CCF with PBFT is in development and should not be used in a production environment.
+
+Raft Consensus Protocol
+-----------------------
+
+We have implemented a version of Raft that provides Crash Fault Tolerance. This means that if we can tolerate up to ``f`` nodes crashing (or being un-responsive) then we need ``2*f + 1`` nodes in total to guarantee liveness. The majority of nodes that need to agree on a change is then ``f + 1``.
+Raft does not make any confidentiality or integrity claims if more than ``f`` nodes have been compromised.
+
+When running CCF with Raft any requests that require the key-value store to be altered go via the Primary node. The Primary node executes the request and then serialises and encrypts the write set resulting from that execution.
```

Comment:
Should we add `Primary` and `Backup` to glossary as, even though they are internal implementation details of the underlying consensus, are exposed to operators (e.g. `getPrimaryInfo` RPC)?

Reply:
We have a mention in the glossary about primaries/backups and leaders/followers. Do you think it needs more explanation there?

Reply:
It would be good to have them defined as terms in the glossary and linked.

Reply:
Good point - I missed it. I think that's fine as it is. 
What about adding a "back" link from the `Consensus Protocols` glossary entry to this new page too? It is likely that users will read the docs in no particular order (if they ever do!).

Reply:
Added link from glossary to consensus page